app/models/poster.rb

```diff
@@ -9,4 +9,40 @@ def self.sort_results(sort)
       Poster.all
     end
   end
+
+  def self.filter_name(name)
+    Poster.where("name ILIKE '%#{name}%'").order(name: :asc)
+  end
+
+  # def self.filter_price(price)
+  #   if price === 'min_price'
+  #     Poster.where("price >= #{price}")
+  #   elsif price === 'max_price'
+  #     Poster.where("price <= #{price}")
+  #   else
+  #     Poster.all
+  #   end
+  # end
+
+  def self.filter_min_price(min_price)
+    Poster.where("price >= #{min_price}")
+  end
+
+  def self.filter_max_price(max_price)
+    Poster.where("price <= #{max_price}")
+  end
+
```

Comment:
Straightforward and clear readability, sql injection is a risk/topic that we should code spike at some point.